lib/charms/prometheus_k8s/v1/prometheus.py

```diff
@@ -279,30 +353,228 @@ def jobs(self):
             its scrape targets.
         """
         scrape_jobs = []
-        for job in self._stored.jobs.values():
-            scrape_jobs.append(json.loads(job))
+
+        for relation in self._charm.model.relations[self._relation_name]:
+            static_scrape_jobs = self._static_scrape_config(relation)
+            if static_scrape_jobs:
+                scrape_jobs.extend(static_scrape_jobs)
 
         return scrape_jobs
 
+    def _static_scrape_config(self, relation):
+        """Generate the static scrape configuration for a single relation.
 
-class PrometheusConsumer(ConsumerBase):
-    _stored = StoredState()
+        Args:
+            relation: an `ops.model.Relation` object whose static
+                scrape configuration is required.
 
-    def __init__(self, charm, name, consumes, multi=False):
-        """Construct a Prometheus charm client.
+        Returns:
+
+            A list (possibly empty) of scrape jobs. Each job is a
+            valid Prometheus scrape configuration for that job,
+            represented as a Python dictionary.
+        """
+        if len(relation.units) == 0:
+            return []
+
+        scrape_jobs = json.loads(relation.data[relation.app].get("scrape_jobs"))
+
+        if not scrape_jobs:
+            return []
+
+        scrape_metadata = json.loads(relation.data[relation.app].get("scrape_metadata"))
+
+        job_name_prefix = "juju_{}_{}_{}_prometheus_{}_scrape".format(
+            scrape_metadata["model"],
+            scrape_metadata["model_uuid"][:7],
```

Comment:
Is model name not unique? If it isn't, we could maybe just check for dups rather than appending a string that is not human readable.

Reply:
The structure of the model name necessarily implies it will be unique for a single Juju controller. It is simpler to ensure that names are unique. This was deemed good enough as far as product requirements go.